apps/cowswap-frontend/src/modules/twap/updaters/TwapOrdersUpdater.tsx

```diff
@@ -89,21 +89,39 @@ export function TwapOrdersUpdater(props: {
 }
 
 function parseOrdersSafeData(ordersSafeData: TwapOrdersSafeData[]): TwapOrderInfo[] {
-  return ordersSafeData.reduce<TwapOrderInfo[]>((acc, data) => {
+  const ordersInfoMap = ordersSafeData.reduce<{ [id: string]: TwapOrderInfo }>((acc, data) => {
     try {
+      const id = getConditionalOrderId(data.conditionalOrderParams)
+      const existingOrder = acc[id]
+
+      /**
+       * There might be two Safe transactions with the same order inside.
+       * But only one of them will be executed.
+       *
+       * For example, you propose a transaction with TWAP order and execute it.
+       * Then, you propose another transaction with the same TWAP order.
+       * After you realize that the proposed transaction is a duplicate, and you replace it or cancel.
+       * In this case we should skip the second transaction, because the first one is already executed.
+       */
+      if (existingOrder?.safeData.safeTxParams.isExecuted) {
+        return acc
```

Comment:
will it be always the first one the good one?

Reply:
There can be only one tx executed for the same nonce.